heron/statemgrs/src/java/com/twitter/heron/statemgr/localfs/LocalFileSystemStateManager.java

```diff
@@ -31,12 +33,50 @@
 import com.twitter.heron.proto.tmaster.TopologyMaster;
 import com.twitter.heron.spi.common.Config;
 import com.twitter.heron.spi.common.Keys;
+import com.twitter.heron.spi.statemgr.Lock;
 import com.twitter.heron.spi.statemgr.WatchCallback;
 import com.twitter.heron.statemgr.FileSystemStateManager;
 
 public class LocalFileSystemStateManager extends FileSystemStateManager {
   private static final Logger LOG = Logger.getLogger(LocalFileSystemStateManager.class.getName());
 
+  /**
+   * Local filesystem implementation of a lock that mimics the file system behavior of the
+   * distributed lock. Only intended for local mode testing, not production. Not guaranteed to be
+   * truly atomic due to limitations in the underlying file system read/write semantics.
+   */
+  private final class FileSystemLock implements Lock {
+    private String path;
+
+    private FileSystemLock(String path) {
+      this.path = path;
+    }
+
+    @Override
```

Comment:
The main use case for locking is the distributed one, where multiple users are trying to make concurrent changes. Since the local file system is only used for compatibility during development by an individual user, do you think it's worth it to invest in a robust local implementation?